Way to open existing Arduino project without creating sample file

Hi,

Thanks for creating this great plugin. I have a small improvement suggestion.
Maybe I did not see the option, but is there a way to open an "exisiting" project and ommit the generation of sample files?
I had an existing project inside a folder, I created a project inside that folder. Then I deleted the sample files (a .cpp file with setup, loop was already existing) and everything worked just fine. I think it would be nice if there is an option to just not add any sample files. What do you think about this?

Also, another small question: What is the purpose of the sloeber.ino.cpp file created? It seems to mimic the primary .ino file a little bit and when I deleted it, the code still compiled without issues.

Kind Regards
Robin

Got following error when trying to build the SDK on Windows 10. Do you have an idea what might cause this?

`
C:\Users\Robin\NoSyncDokumente\arduino-eclipse-plugin>mvn clean verify -PSDK,latest -DskipTests=true
[INFO] Scanning for projects...
[WARNING] No explicit target runtime environment configuration. Build is platform dependent.
[WARNING] No explicit target runtime environment configuration. Build is platform dependent.
[ERROR] Internal error: org.eclipse.tycho.core.ee.UnknownEnvironmentException: Unknown OSGi execution environment: 'JavaSE-12' -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: org.eclipse.tycho.core.ee.UnknownEnvironmentException: Unknown OSGi execution environment: 'JavaSE-12'
`

Maybe I will have more luck on Ubuntu.

--- a/io.sloeber.core/src/io/sloeber/core/api/CodeDescriptor.java
+++ b/io.sloeber.core/src/io/sloeber/core/api/CodeDescriptor.java
@@ -34,7 +34,7 @@ import io.sloeber.core.tools.Stream;
 
 public class CodeDescriptor {
 	public enum CodeTypes {
-		defaultIno, defaultCPP, CustomTemplate, sample
+		None, defaultIno, defaultCPP, CustomTemplate, sample
 	}
 
 	static public final String DEFAULT_SKETCH_BASE = "sketch"; //$NON-NLS-1$
@@ -76,6 +76,10 @@ public class CodeDescriptor {
 
 	private CodeDescriptor(CodeTypes codeType) {
 		myCodeType = codeType;
+	}
+	
+	public static CodeDescriptor createNone() {
+		return new CodeDescriptor(CodeTypes.None);
 	}
 
 	public static CodeDescriptor createDefaultIno() {
@@ -152,6 +156,8 @@ public class CodeDescriptor {
 
 
 		switch (myCodeType) {
+		case None:
+			break;
 		case defaultIno:
 			Helpers.addFileToProject(project, new Path(project.getName() + ".ino"),
 					Stream.openContentStream("/io/sloeber/core/templates/" + DEFAULT_SKETCH_INO, false,replacers),

--- a/io.sloeber.ui/src/io/sloeber/ui/Messages.java
+++ b/io.sloeber.ui/src/io/sloeber/ui/Messages.java
@@ -127,6 +127,7 @@ public class Messages extends NLS {
     public static String ui_new_sketch_arduino_information_help;
     public static String ui_new_sketch_custom_template;
     public static String ui_new_sketch_custom_template_location;
+    public static String ui_new_sketch_none;
     public static String ui_new_sketch_default_cpp;
     public static String ui_new_sketch_default_ino;
     public static String ui_new_sketch_error_failed_to_create_project;

--- a/io.sloeber.ui/src/io/sloeber/ui/messages.properties
+++ b/io.sloeber.ui/src/io/sloeber/ui/messages.properties
@@ -98,6 +98,7 @@ ui_new_sketch_arduino_information=Arduino information
 ui_new_sketch_arduino_information_help=Provide the Arduino information.
 ui_new_sketch_custom_template=Custom template
 ui_new_sketch_custom_template_location=Custom Template Location
+ui_new_sketch_none=No file
 ui_new_sketch_default_cpp=Default cpp file
 ui_new_sketch_default_ino=Default ino file
 ui_new_sketch_error_failed_to_create_project=Failed to create project

--- a/io.sloeber.ui/src/io/sloeber/ui/wizard/newsketch/NewSketchWizardCodeSelectionPage.java
+++ b/io.sloeber.ui/src/io/sloeber/ui/wizard/newsketch/NewSketchWizardCodeSelectionPage.java
@@ -205,6 +205,8 @@ public class NewSketchWizardCodeSelectionPage extends WizardPage {
 	public CodeDescriptor getCodeDescription() {
 
 		switch (CodeTypes.values()[myCodeSourceOptionsCombo.getSelectionIndex()]) {
+		case None:
+			return CodeDescriptor.createNone();
 		case defaultIno:
 			return CodeDescriptor.createDefaultIno();
 		case defaultCPP:
@@ -224,6 +226,8 @@ public class NewSketchWizardCodeSelectionPage extends WizardPage {
 
 	public static String getCodeTypeDescription(CodeTypes codeType) {
 		switch (codeType) {
+		case None:
+			return Messages.ui_new_sketch_none;
 		case defaultIno:
 			return Messages.ui_new_sketch_default_ino;
 		case defaultCPP:

--- a/io.sloeber.ui/src/io/sloeber/ui/wizard/newsketch/NewSketchWizardCodeSelectionPage.java
+++ b/io.sloeber.ui/src/io/sloeber/ui/wizard/newsketch/NewSketchWizardCodeSelectionPage.java
@@ -132,6 +132,11 @@ public class NewSketchWizardCodeSelectionPage extends WizardPage {
 	 */
 	protected void SetControls() {
 		switch (CodeTypes.values()[Math.max(0, myCodeSourceOptionsCombo.getSelectionIndex())]) {
+		case None:
+			myTemplateFolderEditor.setEnabled(false, myParentComposite);
+			myExampleEditor.setEnabled(false);
+			myCheckBoxUseCurrentLinkSample.setEnabled(false);
+			break;
 		case defaultIno:
 			myTemplateFolderEditor.setEnabled(false, myParentComposite);
 			myExampleEditor.setEnabled(false);
@@ -166,9 +171,10 @@ public class NewSketchWizardCodeSelectionPage extends WizardPage {
 			return;
 		}
 		switch (CodeTypes.values()[Math.max(0, myCodeSourceOptionsCombo.getSelectionIndex())]) {
+		case None:
 		case defaultIno:
 		case defaultCPP:
-			setPageComplete(true);// default always works
+			setPageComplete(true);// default and no file always works
 			break;
 		case CustomTemplate:
 			IPath templateFolder = new Path(myTemplateFolderEditor.getStringValue());

